frameworks/Java/grizzly-jersey/src/main/java/hello/DbResource.java

```diff
@@ -29,17 +35,43 @@
   
   @GET
   @Produces(APPLICATION_JSON + "; charset=utf-8")
-  public Object db(@QueryParam("queries") @DefaultValue("1") final int queries) {
+  public Object db(@QueryParam("queries") String queriesParam)
+      throws ExecutionException, InterruptedException {
+
+    final int queries = getQueries(queriesParam);
     final World[] worlds = new World[queries];
     final Random random = ThreadLocalRandom.current();
     final Session session = sessionFactory.openSession();
-    
+    session.setDefaultReadOnly(true);
+    final IdentifierLoadAccess accessor = session.byId(World.class);
+
+    Map<Integer, Future<World>> futureWorlds = new ConcurrentHashMap<>();
+    for (int i = 0; i < queries; i++) {
+      futureWorlds.put(i, Common.EXECUTOR.submit(
+        new Callable<World>() {
+          @Override
+          public World call() throws Exception {
+            return (World) accessor.load(random.nextInt(DB_ROWS) + 1);
```

Comment:
It appears that this method (IdentifierLoadAccess.load) could potentially be returning the data from a cache based off of this description from the [Hibernate JavaDoc](https://docs.jboss.org/hibernate/orm/4.1/javadocs/org/hibernate/IdentifierLoadAccess.html):

> If the instance is already associated with the session, return that instance, initializing it if needed. 

This sounds like it could be a cache. If this is the case, we would need to ensure that it makes direct database access. Do you mind providing some clarification?